src/rdkafka_transport.c

```diff
@@ -498,6 +500,18 @@ static int rd_kafka_transport_ssl_connect (rd_kafka_broker_t *rkb,
 	if (!SSL_set_fd(rktrans->rktrans_ssl, rktrans->rktrans_s))
 		goto fail;
 
+#if (OPENSSL_VERSION_NUMBER >= 0x0090806fL) && !defined(OPENSSL_NO_TLSEXT)
+	/* If non-numerical hostname, send it for SNI */
+	rd_snprintf(name, sizeof(name), "%s", rkb->rkb_nodename);
+	if ((t = strrchr(name, ':')))
```

Comment:
This will truncate an IPv6 address without a port, which probably isnt right (not sure it matters much here though).

The hostname/port is already parsed in broker_add() so I suggest adding a new field to rd_kafka_broker_t:   `char *rkb_hostname;` which you set in broker_add() to the `name` parameter.
And then you can use rkb_hostname directly here.